society-of-biblical-literature-fullnote-bibliography.csl

```diff
@@ -656,42 +828,118 @@
+        <text variable="collection-number" prefix=" "/>
+      </group>
+      <group delimiter=" ">
+        <text term="sub verbo" form="short"/>
+        <text variable="title" text-case="title" quotes="true"/>
+    </group>
+  </group>
+  </macro>
+  <macro name="signed-dictionary-note">
+    <choose>
+      <if position="subsequent">
           <group delimiter=", ">
             <text macro="contributors-short"/>
-            <text macro="title-short"/>
-            <text macro="disambiguate-chapter"/>
-            <text macro="point-locators-subsequent"/>
-          </group>
-        </else-if>
-        <else>
+            <choose>
+              <if position="ibid" match="none">
+                <!-- Excludes Title for ibid entries per CMS 14.34.-->
+                <text macro="title-short"/>
+              </if>
+            </choose>
+          <text macro="point-locators-subsequent"/>
+        </group>
+      </if>
+      <else>
+        <group delimiter="">
           <group delimiter=", ">
             <text macro="contributors-note"/>
             <text macro="title-note"/>
-            <text macro="description-note"/>
-            <text macro="secondary-contributors-note"/>
             <text macro="container-title-note"/>
-            <text macro="container-contributors-note"/>
-            <text macro="locators-note"/>
-            <text macro="collection-title-note"/>
           </group>
-          <text macro="issue-note"/>
-          <text macro="locators-newspaper" prefix=", "/>
           <text macro="point-locators"/>
-          <text macro="access-note" prefix=", "/>
-        </else>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="signed-dictionary-bib">
+    <group delimiter="">
+      <group delimiter=". ">
+        <text macro="contributors"/>
+        <text macro="title-note"/>
+        <text macro="container-title-note"/>
+      </group>
+      <text macro="point-locators"/>
+    </group>
+  </macro>
+  <citation et-al-min="4" et-al-use-first="1" disambiguate-add-names="true">
+    <layout suffix="." delimiter="; ">
+      <choose>
+        <!-- Lexicon/Dictionary/Encyclopedia -->
+        <if type="entry-dictionary entry-encyclopedia" match="any">
+          <choose>
+            <if match="none" variable="author">
+              <!-- Unsigned Lexicon/Dictionary/Encyclopedia -->
+              <text macro="unsigned-dictionary-note"/>
+            </if>
+            <else>
+              <!-- Signed Lexicon/Dictionary/Encyclopedia -->
+              <text macro="signed-dictionary-note"/>
+            </else>
+          </choose>
+        </if>
+        <!-- Not Lexicon/Dictionary/Encyclopedia -->
+            <else-if position="subsequent">
+                  <group delimiter=", ">
+                    <text macro="contributors-short"/>
+                    <!-- Implements conditional check to exclude title in ibid entries (see CMS 14.34) -->
```

Comment:
1) I'd prefer to to nest the ibid check into subsequent simply for readability
2) As per above, I'm worried about just following CMS in their new rule unless we have a very clear indication that's what they want.

Reply:
2. See above re: CMS rules.

1. I'm sorry, I'm not following what you mean. The ibid check is nested inside the subsequent check as far as I can tell, which makes me think I'm not understanding what you mean.

Reply:
If you check how we do this e.g. in CMS:
https://github.com/citation-style-language/styles/blob/master/chicago-fullnote-bibliography.csl#L1310
We first check if ibid/ibid-with-locator, then check if subsequent. What you do is to check for if is subsequent and then differentiate for ibid within that. Both are functionally the same, but all other styles using ibid follow the way this is currently done in CMS, so I'd prefer to keep it that way just for simplicy.

Reply:
Haha, no problem. :) If I understand correct, what I'd need to do is essentially:

If ibid => contributors | locators
If subsequent => contributors | title | locators

This would filter out the ibid exceptions and pass the rest to subsequent. The code would be less DRY, but more readable. Correct?

Reply:
yes, exactly, thanks!

Reply:
Thanks @adam3smith. Fixed this with commit 2969a2a67.